### PR TITLE
doc(context fun): fix example, add some more explanations

### PR DESCRIPTION
**Pull request by robstoll**
Note that the used syntax in the second example actually does not work (I have left a TODO)

### CODE REVIEW

**Comment by robstoll** [comment]
I realised that `given new Table` doesn't compile. Internationally?

**Comment by robstoll** [comment]
I guess I misinterpreted the syntax and ` given T` is actually a special shorthand for `given T = new T()` and not a creator application. Likewise `class A(i : Int); given A(1)` is short for `given A = new A(1)` correct?

if so then I have to update this PR and in this case I would suggest that it is mentioned in http://dotty.epfl.ch/docs/reference/contextual/givens.html

I would do the necessary with an additional commit.

Thoughts?

File: docs/docs/reference/contextual/context-functions.md [hunk 1]
```diff
@@ -81,7 +93,7 @@ with context function types as parameters to avoid the plumbing boilerplate
 that would otherwise be necessary.
 ```scala
   def table(init: Table ?=> Unit) = {
-    given t as Table
+    given t as Table // note the use of a creator application; same as: given t as new Table
```
Comment (odersky) [inline docs/docs/reference/contextual/context-functions.md]:
```suggestion
    given t as Table // note the use of a creator application; same as: given t as Table = new Table
```

**Comment by robstoll** [comment]
@odersky I removed the TODO as the syntax which previously did not work in scastie compiles perfectly fine in master (and now also on scastie):
https://github.com/lampepfl/dotty/pull/8394/commits/659392907db6340a23afe3d5ee4be6c34fb916cc

**Pull request merged**